projects/igniteui-angular/src/lib/services/overlay/overlay.ts

```diff
@@ -933,4 +875,75 @@ export class IgxOverlayService implements OnDestroy {
             this._keyPressEventListener = null;
         }
     }
+
+    private addModalClasses(info: OverlayInfo) {
+        if (info.settings.modal) {
+            const wrapperElement = info.elementRef.nativeElement.parentElement.parentElement;
+            wrapperElement.classList.remove('igx-overlay__wrapper');
+            this.applyAnimationParams(wrapperElement, info.settings.positionStrategy.settings.openAnimation);
+            wrapperElement.classList.add('igx-overlay__wrapper--modal');
+        }
+    }
+
+    private buildAnimationPlayers(info: OverlayInfo) {
+        if (info.settings.positionStrategy.settings.openAnimation) {
+            const animationBuilder = this.builder.build(info.settings.positionStrategy.settings.openAnimation);
+            info.openAnimationPlayer = animationBuilder.create(info.elementRef.nativeElement);
+
+            //  AnimationPlayer.getPosition returns always 0. To workaround this we are getting inner WebAnimationPlayer
+            //  and then getting the positions from it.
+            //  This is logged in Angular here - https://github.com/angular/angular/issues/18891
+            //  As soon as this is resolved we can remove this hack
+            const innerRenderer = (info.openAnimationPlayer as any)._renderer;
+            info.openAnimationInnerPlayer = innerRenderer.engine.players[innerRenderer.engine.players.length - 1];
+            info.openAnimationPlayer.onDone(() => {
+                this.onOpened.emit({ id: info.id, componentRef: info.componentRef });
+                if (info.openAnimationPlayer) {
+                    info.openAnimationPlayer.reset();
+                    // calling reset does not change hasStarted to false. This is why we are doing it here via internal field
+                    (info.openAnimationPlayer as any)._started = false;
```

Comment:
I traced the call to `reset` in the renderer to a `WebAnimationsPlayer.reset` which actually sets `_started` to `false`, is there a situation where a different player is used which doesn't do it?

Reply:
there are some updates in Angular animation library. However there are two things here:
- we can clean up our overlay animations implementation in separate task.
- from the other side we are now in a process of moving overlay animations in separate igxAnimation service when it is not even sure whether or not we will use Angular animations.
This is why I do not think we should fix animations in this PR